File: 백준/Bronze/2231. 분해합/분해합.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String str_N = br.readLine();
        int N_len = str_N.length();

        int N = Integer.parseInt(str_N);
        int res = 0;
        for(int i = (N-(9*N_len)); i < N; i++){
            int num = i;
            int sum = 0;
            while(num != 0){
                sum += num % 10;
                num = num / 10;
            }
            if(sum + i == N){
                res = i;
                break;
            }
        }
        System.out.println(res);
    }
}

File: 백준/Bronze/2231. 분해합/분해합.py
# import sys
# input = sys.stdin.readline
N = int(input())

for i in range(1,N+1):
    num = map(int, str(i))
    num_sum = sum(num) + i
    if num_sum == N:
        print(i)
        break
    if i == N:
        print(0)
# 그러니까 216의 분해합이 198이다. 그러니까 198 + 1 + 9+ 8 이 216이 되는 값을 찾는것임
# 216까지 못돌면 분해합이 없다는것이니 0출력